discovery-infra/test_infra/helper_classes/cluster.py

```diff
@@ -621,6 +621,18 @@ def wait_for_install(self, timeout=consts.CLUSTER_INSTALLATION_TIMEOUT):
             timeout=timeout,
         )
 
+    def _set_dns(self):
+        cluster = self.api_client.cluster_get(self.id)
+        cluster_name = cluster.name
+        base_domain = cluster.base_dns_domain
+        main_cidr = self.nodes.controller.get_machine_cidr()
+        nameserver_ip = str(IPNetwork(main_cidr).ip + 1)
+        fname = f"/etc/NetworkManager/dnsmasq.d/openshift-{cluster_name}.conf"
+        contents = f"server=/api.{cluster_name}.{base_domain}/{nameserver_ip}\n" + f"server=/.apps.{cluster_name}.{base_domain}/{nameserver_ip}\n"
+        self.nodes.controller.tf.change_variables(
+            {"dns_forwarding_file": contents, "dns_forwarding_file_name": fname}
+        )
```

Comment:
What's the reason for managing those files in terraform?

Reply:
+1
Isn't managing it with TF just make it more complex?

Reply:
Terraform treats the file as resource.  It will be destroyed during teardown.

Reply:
For cleanup after use.

Reply:
and yet configuration will still hold until doing ``systemctl NetworkManager reload``, no?
I think it will become hard to investigate network problems

Reply:
It will be destroyed only when **_terraform destroy_** is called.
If you decide to leave the cluster alive, the configuration will remain in place.
During reload, if the cluster was destroyed, the operation will not have any effect (as expected).

Reply:
I think that leaving these files in this directory (part of the global NM) is incorrect behavior.

Reply:
yes, we might become inconsistent either way
anyway, I think it should be ok as is. It should only affect local environments and let's see what will actually happen on those environments

Reply:
I agree with @osherdp it might become a bit complicated to investigate... But we can try. sure, the TF cleanup is a great feature.